File: aoc24/java/src/test/Day06Test.java
import org.junit.jupiter.api.BeforeEach;
import org.junit.jupiter.api.Disabled;
import org.junit.jupiter.api.Test;
import tech.dobler.aoc24.Day06;
import tech.dobler.aoc24.Util;

import static org.assertj.core.api.Assertions.assertThat;

class Day06Test {

    final String testInput = """
            ....#.....
            .........#
            ..........
            ..#.......
            .......#..
            ..........
            .#..^.....
            ........#.
            #.........
            ......#...
            """;
    private Day06 day;

    @BeforeEach
    void setUp() {
        day = new Day06();
    }

    private static String readInput() {
        return Util.readInput(6);
    }

    @Test
    void part1() {
        assertThat(day.part1(testInput)).isEqualTo(41);

        assertThat(day.part1(readInput())).isEqualTo(4665);
    }

    @Test
    void part2() {
        assertThat(day.part2(testInput)).isEqualTo(6);

        assertThat(day.part2(readInput())).isEqualTo(1688);
    }
}

File: aoc21/Kotlin/src/test/kotlin/day06/Day06Test.kt
package day06

import org.junit.jupiter.api.Test
import util.readInput
import kotlin.test.assertEquals

class Day06Test {

    private val path = "06_test"

    @Test
    fun `parse input`() {
        val input = readInput(path, ::mapInput);
        assertEquals(listOf(3, 4, 3, 1, 2), input)
    }

    @Test
    fun `tick once`() {
        assertEquals(mapOf(Pair(0,1L), Pair(1,1), Pair(2,2), Pair(3,1)), tick(mapOf(Pair(1,1), Pair(2,1), Pair(3,2), Pair(4,1))))
    }

    @Test
    fun `tick once with collision`() {
        assertEquals(mapOf(Pair(6,3L), Pair(8,1)), tick(mapOf(Pair(0,1), Pair(7,2))))
    }

    @Test
    fun `tick 18 times`() {
        val input = readInput(path, ::mapInput);
        assertEquals(26, tickNTimes(input, 18))
    }

    @Test
    fun `amount of fish after 18 nticks`() {
        val input = readInput(path, ::mapInput);
        assertEquals(26, tickNTimes(input, 18))
    }

    @Test
    fun `part 1 example`() {
        val input = readInput(path, ::mapInput);
        assertEquals(5934, part1(input))
    }

    @Test
    fun `part 2 example`() {
        val input = readInput(path, ::mapInput);
        assertEquals(26_984_457_539, part2(input))
    }
}